clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/validator/KubernetesValidationUtil.java

```diff
@@ -93,6 +100,25 @@ public boolean validateV2Credentials(AccountCredentialsProvider provider, String
       return false;
     }
 
+    if (!validateNamespace(namespace, (KubernetesV2Credentials)credentials.getCredentials())) {
+      return false;
+    }
+
+    return true;
+  }
+
+  protected boolean validateNamespace(String namespace, KubernetesV2Credentials credentials) {
```

Comment:
I'm thinking we should skip this (e.g. `return true`) when `namespace` is empty. If none is supplied, the default will be used which should be valid.

Reply:
in line 88 we check if the namespace is empty, so it wont reach this part of the code if a namespace is not supplied.

Reply:
Ah I missed that - thanks!

Reply:
Line 89 returns `false`, which is not correct for resources without a namespace. I've submitted https://github.com/spinnaker/clouddriver/pull/2422 to change this.